File: app/src/main/java/com/sanislo/movieapp/domain/movie/MovieRepository.java
package com.sanislo.movieapp.domain.movie;

import android.arch.lifecycle.LiveData;

import com.sanislo.movieapp.domain.model.GenreModel;
import com.sanislo.movieapp.domain.model.MovieModel;
import com.sanislo.movieapp.persistence.response.movieDetails.MovieResponse;

import java.util.List;

import io.reactivex.Single;

public interface MovieRepository {
    Single<MovieResponse> movie(int id);
    LiveData<List<GenreModel>> movieGenresLiveData(int id);
    LiveData<MovieModel> movieLiveData(int id);
}

File: app/src/main/java/com/sanislo/movieapp/domain/movie/MovieRepository.kt
package com.sanislo.movieapp.domain.movie

import android.arch.lifecycle.LiveData

import com.sanislo.movieapp.domain.model.GenreModel
import com.sanislo.movieapp.domain.model.MovieModel
import com.sanislo.movieapp.persistence.response.movieDetails.MovieResponse

import io.reactivex.Single

interface MovieRepository {
    fun movie(id: Int): Single<MovieResponse>
    fun movieGenresLiveData(id: Int): LiveData<List<GenreModel>>
    fun movieLiveData(id: Int): LiveData<MovieModel>
}
